ocaml-lsp-server/test/e2e-new/type_enclosing.ml

```diff
@@ -0,0 +1,491 @@
+open Test.Import
+
+module Util = struct
+  let call_type_enclosing ?(verbosity = 0) ?range_end client position index =
+    let uri = DocumentUri.of_path "test.ml" in
+    let text_document = TextDocumentIdentifier.create ~uri in
+    let params =
+      `Assoc
+        ([ ("textDocument", TextDocumentIdentifier.yojson_of_t text_document)
+         ; ("position", Position.yojson_of_t position)
+         ; ("index", `Int index)
+         ; ("verbosity", `Int verbosity)
+         ]
+        @
+        match range_end with
+        | None -> []
+        | Some x -> [ ("rangeEnd", Position.yojson_of_t x) ])
+    in
+    let params = Some (Jsonrpc.Structured.t_of_yojson params) in
+    let req =
+      Lsp.Client_request.UnknownRequest
+        { meth = "ocamllsp/typeEnclosing"; params }
+    in
+    Client.request client req
+
+  let print_type_enclosing result =
```

Comment:
You can move this to e2e-new/test.ml since its general purpose

Reply:
Is that really general purpose? FMPOV it is really related to testing type enclosing queries.

Reply:
It appears in `e2e-new/completion.ml`, `e2e-new/hover_extended.ml`,  `e2e-new/merlin_call_compatible.ml`.  So I'd say it's general purpose. You gave it a very specific name here, but it could just be called `pretty_print_json`